### PR TITLE
Remove `version` keyword

### LINKED ISSUE #39833: [Improvement]: Remove `version` keyword
### Description

Since we are no longer using the `version` as a keyword, we can remove this restriction from the parser. 

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by ushirask**
Fixes #39833 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by hasithaa — DISMISSED** [review]
Change looks good to me. @rdulmina Can you verify too.

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java [hunk 1]
```diff
@@ -833,111 +825,6 @@ private boolean isEndOfImportDecl(STToken nextToken) {
...
-     **/
-    @Deprecated
-    private void parseVersion() {
-        STToken nextToken = peek();
-        switch (nextToken.kind) {
-            case VERSION_KEYWORD:
-                STNode versionKeyword = parseVersionKeyword();
-                STNode versionNumber = parseVersionNumber();
-                addInvalidNodeToNextToken(versionKeyword,
-                        DiagnosticErrorCode.ERROR_VERSION_IN_IMPORT_DECLARATION_NO_LONGER_SUPPORTED);
```
Comment (rdulmina) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java:851]: Remove this from `DiagnosticErrorCode.java` as well

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java [hunk 2]
```diff
@@ -44,7 +44,7 @@ public void test(String fileName, String path) throws IOException {
 //        Path buildDirectory = Paths.get("build").toAbsolutePath().normalize();
 //
 //        return new Object[][] {
-//                {"annot_decl_source_01.bal", getFilePath("annot_decl_source_01.bal",
+//                {"import_decl_source_08.bal", getFilePath("import_decl_source_08.bal",
```
Comment (rdulmina) [inline misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java:47]: Do we need this change

File: compiler/ballerina-parser/src/test/resources/misc/identifiers/version_as_identifier_source.bal [hunk 3]
```diff
@@ -0,0 +1,3 @@
+public function func() {
+    int version = 3;
```
Comment (rdulmina) [inline compiler/ballerina-parser/src/test/resources/misc/identifiers/version_as_identifier_source.bal:2]: shall we add some more cases?
```
version a = 3;
int b = version;
```

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java (hunk 2 above)
Comment (ushirask) [reply misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java:47]: Changed in testing, will revert.

**Review by rdulmina — APPROVED** [review]

**Review by hasithaa — APPROVED** [review]

**Review by rdulmina — APPROVED** [review]

**Pull request merged**